MLIR.cmake

```diff
@@ -25,16 +25,55 @@ else()
 endif()
 
 # LLVM project lib folder
-set(LLVM_PROJECT_LIB ${LLVM_PROJ_BUILD}/lib)
+if (ENV{LLVM_PROJECT_LIB})
+  set(LLVM_PROJECT_LIB $ENV{LLVM_PROJECT_LIB})
+else()
+  if(MSVC)
+    if (CMAKE_BUILD_TYPE)
+      set(LLVM_PROJECT_LIB ${LLVM_PROJ_BUILD}/${CMAKE_BUILD_TYPE}/lib)
+    else()
+      set(LLVM_PROJECT_LIB ${LLVM_PROJ_BUILD}/release/lib)
+    endif()  
+  else()
+    set(LLVM_PROJECT_LIB ${LLVM_PROJ_BUILD}/lib)
+  endif()
+endif()
+message(STATUS "LLVM_PROJECT_LIB:" ${LLVM_PROJECT_LIB})
+
+# LLVM project bin folder
+if (ENV{LLVM_PROJ_BIN})
+  set(LLVM_PROJ_BIN $ENV{LLVM_PROJ_BIN})
+else()
+  if(MSVC)
+    if (CMAKE_BUILD_TYPE)
+      set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/${CMAKE_BUILD_TYPE}/bin)
+    else()
+      set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/Release/bin)
+    endif()
+  else()
+    set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/bin)
+  endif()
+endif()
+message(STATUS "LLVM_PROJ_BIN:" ${LLVM_PROJ_BIN})
 
 # Include paths for MLIR
 set(LLVM_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/llvm/include)
 set(LLVM_BIN_INCLUDE_PATH ${LLVM_PROJ_BUILD}/include)
 set(MLIR_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/mlir/include)
 set(MLIR_BIN_INCLUDE_PATH ${LLVM_PROJ_BUILD}/tools/mlir/include)
-set(MLIR_TOOLS_DIR ${LLVM_PROJ_BUILD}/bin)
+set(MLIR_TOOLS_DIR ${LLVM_PROJ_BIN})
 
-set(ONNX_MLIR_TOOLS_DIR ${ONNX_MLIR_BIN_ROOT}/bin)
+# ONNX-MLIR tools folder
+if(MSVC)
+  if (CMAKE_BUILD_TYPE)
+    set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/${CMAKE_BUILD_TYPE})
+  else()
+    set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/Release)
```

Comment:
onnx-mlir's path of bin\Release is the default under CMake, LLVM overrides this. 
We could take llvm's path - it would just involve touching more CMake files than what's in this change.
It's mostly a preference for how you want to ship binaries. There are pluses and minuses to both approaches: the first groups all binaries for the same build config under the config directory i.e. for a particular build config like Release, you can have something like outputdir\Release\lib\Foo, outputdir\Release\lib\Bar. Now all Release libs are grouped together and it's easy to deploy one build config at a time.
The disadvantage is that now files for the same library are under 2 different top-level paths in the tree. If you wanted to keep binaries of different build configs together, you'd go the default CMake route i.e. outputdir\lib\Foo\Debug,  outputdir\lib\Foo\Release. Now all of the different build config binaries for Foo are together under the Foo subdirectory. This can be useful if you want to ship multi-config packages for certain libraries e.g. ones that have Debug and Release flavors together.